packages/cli/src/pull/beta.ts

```diff
@@ -37,47 +37,52 @@ export type PullOptionsBeta = Required<
 export async function handler(options: PullOptionsBeta, logger: Logger) {
   const { OPENFN_API_KEY, OPENFN_ENDPOINT } = process.env;
 
-  const config: Partial<Config> = {
+  const cfg: Partial<Config> = {
     apiKey: options.apiKey,
     endpoint: options.endpoint,
   };
 
   if (!options.apiKey && OPENFN_API_KEY) {
     logger.info('Using OPENFN_API_KEY environment variable');
-    config.apiKey = OPENFN_API_KEY;
+    cfg.apiKey = OPENFN_API_KEY;
   }
 
   if (!options.endpoint && OPENFN_ENDPOINT) {
     logger.info('Using OPENFN_ENDPOINT environment variable');
-    config.endpoint = OPENFN_ENDPOINT;
+    cfg.endpoint = OPENFN_ENDPOINT;
   }
 
+  // TODO `path` or `output` ?
+  // I don't think I want to model this as output. deploy is really
+  // designed to run from the working folder
+  // could be projectPath or repoPath too
+  const outputRoot = path.resolve(options.path || '.');
+
+  // TODO is outputRoot the right dir for this?
+  const workspace = new Workspace(outputRoot);
+  const config = workspace.getConfig();
+
   // download the state.json from lightning
-  const { data } = await getProject(config as DeployConfig, options.projectId);
+  const { data } = await getProject(cfg as DeployConfig, options.projectId);
 
   // TODO if the user doesn't specify an env name, prompt for one
   const name = options.env || 'project';
 
   const project = Project.from('state', data, {
-    endpoint: config.endpoint,
+    repo: config,
+    endpoint: cfg.endpoint,
     env: name,
     fetched_at: new Date().toISOString(),
   });
 
-  // TODO `path` or `output` ?
-  // I don't think I want to model this as output. deploy is really
-  // designed to run from the working folder
-  // could be projectPath or repoPath too
-  const outputRoot = path.resolve(options.path || '.');
-
   const projectFileName = project.getIdentifier();
 
   await fs.mkdir(`${outputRoot}/.projects`, { recursive: true });
   let stateOutputPath = `${outputRoot}/.projects/${projectFileName}`;
 
   const workflowsRoot = path.resolve(
     outputRoot,
-    project.repo?.workflowRoot ?? 'workflows'
+    project.repo.dirs?.workflows ?? 'workflows'
```

Comment:
Shouldn't need any of this optional chaining bullshit: I want good robust config with everything defaulted